src/likelihoods/negativebinomial.jl

```diff
@@ -1,36 +1,170 @@
+abstract type NBParam end
+default_invlink(::NBParam) = error("Not implemented")
+
+abstract type NBParamProb <: NBParam end
+default_invlink(::NBParamProb) = logistic
+abstract type NBParamMean <: NBParam end
+default_invlink(::NBParamMean) = exp
+
 """
-    NegativeBinomialLikelihood(l=logistic; successes::Real=1)
+    NegativeBinomialLikelihood(param::NBParam, l::Function/Link)
```

Comment:
```suggestion
    NegativeBinomialLikelihood(param::NBParam, link::Function/Link)
```
that's what you call it further down in the docstring (and I think it's a more helpful variable name anyways) :)

also, how about
```suggestion
    NegativeBinomialLikelihood(param::NBParam, link::Union{Function,Link})
```
?